products/bridge/test/utils.ts

```diff
@@ -186,18 +256,15 @@ export async function dispatchCall(
 
   // other vaidators see the Dispatched event and retrieve the result of the relayed call
   const results = await Promise.all(
-    signerIndices.map(async (index) => {
-      const blockNum = await ethers.provider.getBlockNumber();
+    validators.map(async (validator) => {
       const filter = relayer.filters.Dispatched(
         caller,
         undefined,
         undefined,
         undefined,
         nonce
       );
-      const logs = await relayer
-        .connect(validators[index])
-        .queryFilter(filter, blockNum - 100, blockNum);
+      const logs = await relayer.connect(validator).queryFilter(filter);
```

Comment:
As discussed, it's essential to only consider events in finalized blocks, therefore the code before refactoring fetched `ethers.provider.getBlockNumber('finalized')` and queried only until that block number.